kotlin/workflow-ui-android/src/main/java/com/squareup/workflow/ui/WorkflowFragment.kt

```diff
@@ -0,0 +1,99 @@
+/*
+ * Copyright 2019 Square Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.squareup.workflow.ui
+
+import android.arch.lifecycle.ViewModelProviders
+import android.os.Bundle
+import android.support.v4.app.Fragment
+import android.support.v4.app.FragmentActivity
+import android.view.LayoutInflater
+import android.view.View
+import android.view.ViewGroup
+import com.squareup.workflow.Workflow
+import io.reactivex.Flowable
+
+@ExperimentalWorkflowUi
+abstract class WorkflowFragment<InputT, OutputT : Any> : Fragment() {
```

Comment:
I'm not a fan of base classes, but I also don't see how to do better here.

Reply:
Same. I used a support class for the activity use case, but for fragments this seemed like the most natural fit. And at least the bulk of it is in the view model.